editor/src/clj/editor/settings_core.clj

```diff
@@ -175,39 +177,51 @@
    :comma-separated-list []
    :string ""
    :boolean false
-   :integer 0
-   :number 0.0})
-
-(defn- add-type-defaults [meta-info]
-  (update-in meta-info [:settings]
-             (partial map (fn [setting] (update setting :default #(if (nil? %) (type-defaults (:type setting)) %))))))
+   :integer (int 0) ; Parsed as 32-bit integer in the engine runtime.
+   :number (float 0.0)}) ; Parsed as 32-bit float in the engine runtime.
+
+(defn- sanitize-default [default type]
+  (case type
+    :integer (int default) ; Parsed as 32-bit integer in the engine runtime.
+    :number (float default) ; Parsed as 32-bit float in the engine runtime.
+    default))
+
+(defn- ensure-type-defaults [meta-info]
+  (update meta-info :settings
+          (fn [settings]
+            (mapv (fn [{:keys [type] :as meta-setting}]
+                    (update meta-setting :default
+                            (fn [default]
+                              (if (some? default)
+                                (sanitize-default default type)
+                                (type-defaults type)))))
+                  settings))))
```

Comment:
Instead of just adding missing type defaults, we now sanitize the defaults to ensure they have the correct numeric types.